src/pip/_internal/vcs/git.py

```diff
@@ -141,6 +145,7 @@ def get_revision_sha(cls, dest, rev):
             pass
 
         refs = {}
+        sha = None
         for line in output.strip().splitlines():
             try:
                 sha, ref = line.split()
```

Comment:
It might be better to please mypy by renaming this `sha` variable instead of setting it to None before, to reflect the fact that it is not the same variable as the one used later in `sha = refs.get(branch_ref)`.

Reply:
Thanks for the suggestion. Done in the latest revision.